ThunderBasics/MultipleShadowView.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  MultipleShadowView.swift
+//  ShadowView
+//
+//  Created by Simon Mitchell on 23/07/2020.
+//  Copyright © 2020 3 Sided Cube. All rights reserved.
+//
+
+import UIKit
+
+
+protocol CornerObservableLayerDelegate: class {
+    
+    func cornerObservableLayer(_ layer: CornerObservableLayer, didUpdateCornerRadius cornerRadius: CGFloat)
+    
+    func cornerObservableLayer(_ layer: CornerObservableLayer, didUpdateCornerCurve cornerCurve: CALayerCornerCurve)
+}
+
+internal class CornerObservableLayer: CALayer {
+    
+    /// Delegate for corner property changes
+    weak var cornerDelegate: CornerObservableLayerDelegate?
+    
+    override var cornerRadius: CGFloat {
+        didSet {
+            cornerDelegate?.cornerObservableLayer(self, didUpdateCornerRadius: cornerRadius)
+        }
+    }
+    
+    override var cornerCurve: CALayerCornerCurve {
+        didSet {
+            cornerDelegate?.cornerObservableLayer(self, didUpdateCornerCurve: cornerCurve)
+        }
+    }
+}
+
+/// A subclass of `UIView` which ensures multiple shadows have their
+/// bounds, and corner radius properties updated when the parent view's properties
+/// change.
+public class MultipleShadowView: UIView {
+    
+    public override class var layerClass: AnyClass {
+        return CornerObservableLayer.self
+    }
+    
+    /// Whether the shadow layers should update their corner radius to match the view's
+    /// layer's corner radius
+    public var matchShadowCornerRadius: Bool = true {
+        didSet {
+            guard matchShadowCornerRadius else { return }
+            layer.sublayers?.filter({ $0 is ShadowLayer }).forEach({ (shadowLayer) in
```

Comment:
Great idea! Big fan, but let's get this into a function rather than duplicating it in `layoutSubviews` otherwise a future change might miss that they need to change both

Reply:
Yep, had a feeling you may say something along those lines ;)